fil_fungible_token/src/runtime/messaging.rs

```diff
@@ -94,56 +91,64 @@ impl Messaging for FvmMessenger {
-/// If Address is actor-address or bls-address, it returns an error simulating an uninitialised address
-///
-/// # initialise_account
-/// Simulates the initialisation of an account
-/// - ID, SECP addresses panic as they should not be called given that resolve_id gave back an ActorID
-/// - BLS addresses give back FAKE_INITIALIZED_ID
-/// - Actor addresses are uninitialised and give back an error
 #[derive(Debug)]
 pub struct FakeMessenger {
+    pub last_hook: RefCell<Option<TokenReceivedParams>>,
     address_resolver: RefCell<FakeAddressResolver>,
+    actor_id: ActorID,
+    abort_next_send: RefCell<bool>,
 }
 
+/// A mocked messenger that can be used to interact with other Actors
+///
+/// Can be used to test behaviour when other Actors abort when handling messages
 impl FakeMessenger {
     /// Creates a new FakeMessenger with a given set of initialized accounts
     ///
     /// first_usable_actor_id is the first ActorID that has not been already allocated to an address
     /// i.e. in test fixtures where it may be useful to have statically allocated ID addresses, they
     /// should all have an ActorID strictly below first_usable_actor_id
-    pub fn new(first_usable_actor_id: ActorID) -> Self {
-        Self { address_resolver: RefCell::new(FakeAddressResolver::new(first_usable_actor_id)) }
+    pub fn new(actor_id: ActorID, first_usable_actor_id: ActorID) -> Self {
+        Self {
+            actor_id,
+            address_resolver: RefCell::new(FakeAddressResolver::new(first_usable_actor_id)),
+            last_hook: Default::default(),
+            abort_next_send: RefCell::new(false),
+        }
+    }
+
+    pub fn abort_next_send(&mut self) {
+        self.abort_next_send.replace(true);
     }
 }
 
 impl Messaging for FakeMessenger {
-    fn call_receiver_hook(
+    fn actor_id(&self) -> ActorID {
+        self.actor_id
+    }
+
+    fn send(
         &self,
-        _from: ActorID,
-        _to: ActorID,
+        _to: &Address,
+        method: MethodNum,
+        params: &RawBytes,
         _value: &TokenAmount,
-        data: &[u8],
     ) -> Result<Receipt> {
-        if data.is_empty() {
-            Ok(Receipt {
-                exit_code: ExitCode::OK,
-                return_data: RawBytes::new(data.to_vec()),
+        if *self.abort_next_send.borrow() {
+            self.abort_next_send.replace(false);
+            return Ok(Receipt {
+                exit_code: ExitCode::USR_UNSPECIFIED,
                 gas_used: 0,
-            })
-        } else {
-            Ok(Receipt {
-                exit_code: ExitCode::SYS_INVALID_RECEIVER,
-                return_data: RawBytes::new(data.to_vec()),
-                gas_used: 0,
-            })
+                return_data: Default::default(),
+            });
+        }
+
+        if method == RECEIVER_HOOK_METHOD_NUM {
+            let hook_params = params.deserialize().unwrap();
+            self.last_hook.borrow_mut().replace(hook_params);
         }
```

Comment:
else fail? whatever method num was called probably doesn't exist (until we add more mocking ability)